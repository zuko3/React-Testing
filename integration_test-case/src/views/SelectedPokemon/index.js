import React, { Component } from "react";
import { connect } from "react-redux";
import history from "../../history";

class SelectedPokemon extends Component {
  componentDidMount() {
    const { name } = this.props;

    if (!name) {
      history.replace("/");
    }
  }

  render() {
    const { name, img } = this.props;
    return (
      <div>
        <h2>Congrats!</h2>
        <h3>You have selected {name}!</h3>
        <img src={img} alt={name} />
      </div>
    );
  }
}

export default connect(state => {
  const selectedPokemonId = state.session.selectedPokemonId;
  const activePokemon = state.data.pokemon[selectedPokemonId] || {
    species: {},
    sprites: {}
  };

  return {
    name: activePokemon.species.name,
    img: activePokemon.sprites.front_default
  };
})(SelectedPokemon);
