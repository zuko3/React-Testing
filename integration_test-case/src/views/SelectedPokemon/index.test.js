import {
  makeMountRender,
  mockData,
  reduxify,
  snapshotify
} from "../../test-utils";
import SelectedPokemon from "./";

describe("<SelectedPokemon />", function() {
  it("matches snapshot", function() {
    const mockPokemonData = mockData.pokemon;
    const mockPokemonId = mockPokemonData.id;

    const wrapper = makeMountRender(
      reduxify(
        SelectedPokemon,
        {},
        {
          data: { pokemon: { [mockPokemonId]: mockData.pokemon } },
          session: { selectedPokemonId: mockPokemonId }
        }
      )
    );

    expect(snapshotify(wrapper)).toMatchSnapshot();
  });

  it("redirects to homepage if there is no selected pokemon", function() {
    const wrapper = makeMountRender(reduxify(SelectedPokemon));

    wrapper.update();
    expect(window.location.href).toBe("http://localhost/");
  });
});
