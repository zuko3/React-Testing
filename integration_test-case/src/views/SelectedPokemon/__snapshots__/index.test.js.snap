// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectedPokemon /> matches snapshot 1`] = `"<div><h2>Congrats!</h2><h3>You have selected !</h3><img></div>"`;
